src/python_testing/TC_RVCCLEANM_1_2.py

```diff
@@ -0,0 +1,121 @@
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_RVCCLEANM_1_2(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.RvcCleanMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    @async_test_body
+    async def test_TC_RVCCLEANM_1_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        attributes = Clusters.RvcCleanMode.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        if self.check_pics("RVCCLEANM.S.A0000"):
+            self.print_step(2, "Read SupportedModes attribute")
+            supported_modes = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes))
+
+            asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least two entries!")
+
+            modes = []
+            for m in supported_modes:
+                if m.mode in modes:
+                    asserts.fail("SupportedModes must have unique mode values!")
+                else:
+                    modes.append(m.mode)
+
+            labels = []
+            for m in supported_modes:
+                if m.label in labels:
+                    asserts.fail("SupportedModes must have unique mode label values!")
+                else:
+                    labels.append(m.label)
+
+                # common mode tags
+                commonTags = {0x0: 'Auto',
+                              0x1: 'Quick',
+                              0x2: 'Quiet',
+                              0x3: 'LowNoise',
+                              0x4: 'LowEnergy',
+                              0x5: 'Vacation',
+                              0x6: 'Min',
+                              0x7: 'Max',
+                              0x8: 'Night',
+                              0x9: 'Day'}
+
+                # RVC Run mode tags
+                # runTags = {0x4000: 'Idle',
+                #           0x4001: 'Cleaning'}
+
+                # RVC Clean mode tags
+                cleanTags = {0x4000: 'DeepClean',
+                             0x4001: 'Vacuum',
+                             0x4002: 'Mop'}
+
+            for m in supported_modes:
+                for t in m.modeTags:
```

Comment:
Is it possible for a mode option to have the same tags as an other? The test plan does not specify the need for this check but I am wondering if it was missed and we should raise an issue to add it.